gaphor/UML/sanitizerservice.py

```diff
@@ -54,10 +57,23 @@ def _unlink_on_subject_delete(self, event):
         """Unlink the model element if no more presentations link to the
         `item`'s subject or the deleted item was the only item currently
         linked."""
-        if event.property is Presentation.subject:  # type: ignore[misc]
-            old_subject = event.old_value
-            if old_subject and not old_subject.presentation:
-                old_subject.unlink()
+        if (
+            not self.properties.get("remove-dangling-elements", True)
+            or event.property is not Presentation.subject  # type: ignore[misc]
+        ):
+            return
+
+        old_subject = event.old_value
+        if old_subject and not old_subject.presentation:
+            old_subject.unlink()
+
+            self.event_manager.handle(
+                Notification(
+                    gettext(
+                        "Dangling model element(s) have been removed: they're not shown in a diagram anymore."
```

Comment:
Maybe the wording can be improved.

Reply:
What about: Since you deleted the last usage of elements in a diagram, they  were automatically removed from the model.

Reply:
That sounds friendly. Let me try that.

![image](https://user-images.githubusercontent.com/96249/146407119-9d460ef9-cc31-4fee-b481-990ec0d1c48b.png)

Can it be a little shorter?

"You deleted the last usage of elements, they are removed from the model too."?

"You deleted the last usage of elements in a diagram, they are removed from the model too."?

Reply:
I like the last one, I think it is short and clear.

Reply:
"Removed model elements no longer in use in a diagram." it is then.

Reply:
As I user i would like to understand what has happened in the system. The description "Removed model elements no longer in use in a diagram" is something I don't understand as a user.

"The deleted element was removed from the model as well because it was not used in any other diagram"?

Reply:
The message should ideally be short, since it disappears after a couple of seconds and you need time to read it. I agree that @sz332's and @danyeaw's initial phrasing is more self explaining (less cryptic if you like).

This message occurs in two situations:
1. An element has been removed that's no longer in any diagram
2. A diagram has been removed that contains elements that are now no longer present in any diagram.